uwblueprint/childrens-aid-society
# Caregiver CaseOverview component

## Notion ticket link

[Case Overview Visiting Family component](https://www.notion.so/uwblueprintexecs/134e1f159caa4f73939d11a700c325ad?v=39cf7d5a98384eb9af2bff2265d0ac8a&p=f88ca4abaa9d45888ac5433a3cc75074&pm=s)



## Implementation description
frontend:
- New CasePromptBox component to display each caseoverview components details
- New CaregiverAPIClient for frontend/backend integration for Caregiver APIs (PUT and POST)
- Existing NewCaregiverModal component is edited to handle both POST and PUT calls
- Existing NewCaregiverModal displays existing caregiver data if a caregiver is passed in
- CaseOverview page pulls Caregiver data from backend through `get_caregivers_by_intake_id`
- New CaregiverDetails and CaregiverRelationship types

backend:
- work in `caregiver_service` functions for editing existing caregiver, `update_caregiver`
- need to add to `caregiver_routes` for routes to PUT (edit)
- functional tests for `get_caregiver_by_intake_id` and `update_caregiver`
- `caregiver.py` model edited to add enum values for `caregiver_relationship_to_child`
- new migration made for changes ^
- regression changes for change in enum values in db_seed


## Steps to test
1. postman to check API routes 
2. pytest to run functional tests
3. click on different cases in homepage to view different lists of caregivers for each intake_id
4. make dummy caregivers in the frontend through caseoverview
5. edit dummy caregivers in frontend



## What should reviewers focus on?
* added new button style in the buttonstyles.tsx file
* lmk if things need to be refactored (ik state is looking to be refactored, I can do that in a later ticket)


## Checklist
